src/toil/test/batchSystems/batchSystemTest.py

```diff
@@ -577,6 +577,118 @@ class SingleMachineBatchSystemJobTest(hidden.AbstractBatchSystemJobTest):
+        that can run.
+        """
+        tempDir = self._createTempDir('testFiles')
+
+        options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+        options.workDir = tempDir
+        options.maxCores = 4
+        from toil import physicalMemory
+        availableMemory = physicalMemory()
+        options.batchSystem = self.batchSystemName
+
+        outFile = os.path.join(tempDir, 'counter')
+        open(outFile, 'w').close()
+
+        root = Job()
+        # writes A
+        blocker = Job.wrapFn(_resourceBlockTestAuxFn, outFile=outFile, sleepTime=30, writeVal='A',
+                             cores=2, memory='1M', disk='1M')
+        firstJob = Job.wrapFn(_resourceBlockTestAuxFn, outFile=outFile, sleepTime=5, writeVal='A',
+                              cores=1, memory='1M', disk='1M')
+        secondJob = Job.wrapFn(_resourceBlockTestAuxFn, outFile=outFile, sleepTime=10, writeVal='A',
+                               cores=1, memory='1M', disk='1M')
+        # Writes B
+        # Should block off 50% of memory while waiting for it's 3 cores
+        firstJobChild = Job.wrapFn(_resourceBlockTestAuxFn, outFile=outFile, sleepTime=0,
+                                   writeVal='B', cores=3, memory=(availableMemory/2), disk='1M')
+        # Writes C
+        # These two shouldn't be able to run before B because there should be only
+        # (50% of memory - 1M) available (firstJobChild should be blocking 50%)
+        secondJobChild = Job.wrapFn(_resourceBlockTestAuxFn, outFile=outFile, sleepTime=5,
+                                    writeVal='C', cores=2, memory=(availableMemory/1.5), disk='1M')
+        # writes D
+        secondJobChildChild = Job.wrapFn(_resourceBlockTestAuxFn, outFile=outFile, sleepTime=5,
+                                         writeVal='D', cores=2, memory=(availableMemory/1.5),
+                                         disk='1M')
+
+        root.addChild(blocker)
+        root.addChild(firstJob)
+        root.addChild(secondJob)
+
+        firstJob.addChild(firstJobChild)
+        secondJob.addChild(secondJobChild)
+
+        secondJobChild.addChild(secondJobChildChild)
+        """
+        The tree is:
+                    root
+                  /   |   \
+                 bl   fJ   sJ
+                      |    |
+                      fJC  sJC
+                           |
+                           sJCC
+        But the order of execution should be
+        root > bl, fJ, sJ > sJC > sJCC > fJC
+        since fJC cannot run till bl finishes but sJC and sJCC can(fJC blocked by disk). If the
+        resource acquisition is written properly, then fJC which is scheduled before sJC and sJCC
+        should not block them, and should only run after they finish.
+        """
+        Job.Runner.startToil(root, options)
+        with open(outFile) as oFH:
+            outString = oFH.read()
+        self.assertEqual(outString, 'AAACDB')
```

Comment:
Can you make the writeVal values consistent with the job name? That way I won't have to hunt down each job's writeVal to verify your assertion against what you write in the comment above.